tasks/internal/update-fbc-catalog-task/update-fbc-catalog-task.yaml

```diff
@@ -272,6 +273,8 @@ spec:
                 esac
             done
             echo
+            # buildState is a json string that should be escaped to garantee that there is no evaluation of the content
+            # will happen before it is read by the calling task.
```

Comment:
First off, the string needs some tweaks "there is no evaluation .. will happen" - probably ".. that no evaluation of the content happens ..."

But also, didn't we just come to the conclusion in another PR that it doesn't need to be escaped at all and it will be fixed later? Is it still your intention to fix it later? Or should it be fixed in this PR?

Reply:
the problem is that the escaped results will consumed by the calling task/pipeline.  If the results are unescaped, I don't know how the pipeline would behave - I am pretty sure it will break. As of last week, I could not yet have a cluster to fully test the fbc pipeline , so I would need to wait for a fully e2e run (who knows how many times) just for a fix that is not really necessary. 

I personally think we don't need that change as your other suggestions were enough to not need to remove the shellcheck disables, but I am open to listen if there is any other good reason.

Reply:
I don't see a reason why it shouldn't work with setting a env variable from the parameter as suggested in the other PR. And that's something you can easily confirm/test in a task test. 

But I am fine keeping it as is, just with some changed wording here, like "... needs to be escaped because of the way it is consumed in another task."

Reply:
> I don't see a reason why it shouldn't work with setting a env variable from the parameter as suggested in the other PR. And that's something you can easily confirm/test in a task test.

The problem is testing the fbc-pipeline itself (add-fbc-contribution), that reads the buildState and JsonResults, but they are sent through the InternalServices Controller, so that's why I would need to run at least the add-fbc-contribution to confirm that. 

I will give it a try - I just did not want to miss the wednesday promotion in case it takes too long. If it does, I will leave it this way (but changing the text).

Reply:
@mmalina I changed the line in the update-fbc-catalog task to not escape the json. It worked running locally. Lets hope all goes ok in e2e.

Reply:
ok, cool, thanks